electron/electron/handlers/runWhisper/runWhisper.ts

```diff
@@ -100,34 +108,36 @@ export default ipcMain.handle(
         'poetry',
         ['run', 'stagewhisper', ...inputArray],
         {
-          stdio: 'inherit',
+          // stdio: 'pipe',
+          env,
           cwd: backendDir
         }
       ];
     } else {
       childProcessArgs = ['echo', ['"Production backend not yet implemented"']];
     }
-    const childProcess = spawn(...childProcessArgs);
-
     const transcription = await new Promise<Transcription>((resolve, reject) => {
-      childProcess.on('error', (error: Error) => {
-        console.log(`RunWhisper: Failed to start subprocess: ${error}`);
-      });
+      const childProcess = spawn(...childProcessArgs);
 
-      // If the stdio/out/err streams are available, read them to the console
-      // Note: if `stdio` is set to anything but `"pipe"` in the `spawn` opts,
-      // then these will be undefined and these events will never fire.
-      childProcess.stdout?.on('data', (data: string) => {
-        console.log(`stdout: ${data}`);
+      // Detect if parent node process is killed
+      process.on('SIGINT', () => {
+        console.log('RunWhisper: Parent process killed, killing child process');
+        childProcess.kill();
+        process.exit();
       });
-      childProcess.stdin?.on('data', (data: string) => {
-        console.log(`stdin: ${data}`);
+
+      // Create a line reader to read the output of the whisper script
+      const lineReader = readline.createInterface({
+        input: childProcess.stdout,
+        output: childProcess.stdin,
```

Comment:
Are these correct? I don't actually fully understand what the line reader does, so they might be, but I wanted to double check

Reply:
We could remove the output value, this interface takes the output of the python script as its input and offers a way to pass data back via the output channel. Ill remove the output channel and re-commit